src/clusterfuzz/_internal/build_management/overrides.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions to apply build and revisions path overrides"""
+
+import ast
+import os
+
+from clusterfuzz._internal.google_cloud_utils import storage
+from clusterfuzz._internal.system import environment
+
+PLATFORM_ID_URLS_FILENAME = 'config.json'
+PLATFORM_TO_BUILD_PATH_KEY = 'build_paths'
+PLATFORM_TO_REV_PATH_KEY = 'revisions_paths'
+
+
+def check_and_apply_overrides(curr_path, config_key):
+  """Check if the given file points to a config, if so, use that to override
+  any given paths"""
+  if not curr_path:
+    return curr_path
+  if os.path.basename(curr_path) == PLATFORM_ID_URLS_FILENAME:
+    curr_path = _apply_platform_id_overrides(environment.get_platform_id(),
+                                             curr_path, config_key)
+  return curr_path
+
+
+def _apply_platform_id_overrides(platform_id, config_url, config_key):
+  """read the `bucket_path`, parse as JSON, and map based on platform_id."""
+  new_path = config_url
+  config_dict = _get_config_dict(config_url)
+  found, path = _get_path_from_config(config_dict, config_key, platform_id)
```

Comment:
If we get to this point and there's no path found, that seems like a fatal error. Just exception out here. 

Define our own exception type:

```
class BuildOverrideError(Exception):
  """Build override error."""


raise BuildOverrideError('Could not find correct build override path.')
```